### ISSUE TITLE
Make buildkite testing pipeline for dmd treat deprecations as errors

### ISSUE THREAD

**Issue by RazvanN7** [dlang/project-ideas#105]
## Description

Once a partic ular feature is deprecated,  tested projects that are using that feature are still going to successfully pass the testing pipeline - buildkite in our case. This doesn't indicate what is the level of disruption that a particular deprecation yields and may lead to issuing a "blizzard of deprecations" that fall under the radar for the person that does the deprecation.

The purpose of this project is make buildkite treat deprecations as errors and then update all of the currently tested projects to not issue any deprecations.

## What are rough milestones of this project?

- Create PR for buildkite to treat deprecations as warnings in the dmd testing pipeline
- Fix each project that fails as a consequence of the changes in the first step

## How does this project help the D community?

The most important D projects should be present in the buildkite testing pipeline. Therefore, when introducing a deprecation we can get an idea of what the actual "breakage" is going to be if a feature is to be dropped. In addition, fixing the most popular projects to not issue any deprecations is going to make it easier and far less confusing to use those packages.

## Recommended skills

Git, D, perseverence, patience, make, bash, CI/CD

## Point of Contact

@RazvanN7

**Comment by Geod24** [comment]
Hardly workable. Look at Ocean, it explicitly treat deprecations as such (because the default is deprecation as error).
It should be the responsibility of the person introducing the deprecation to do this work. It's as simple as switching a boolean on in the code, see what breaks, and fix it. But you might have to wait for new releases for a deprecation to be mergeable.

**Comment by RazvanN7** [comment]
I agree it should be the responsibility of the person introducing the deprecation, but since the testing pipeline passes successfully how is one supposed to see and fix the errors? If the answer is: "manually check each project", that's not realistic. Therefore, if we realistically want to avoid having deprecations forever in the projects tested by buildkite, we need to fix them now and enable -de for all of them.
